### DISCUSSION TITLE
Is there a way to delete files from a backup?

### DISCUSSION THREAD

**Discussion by benjamindaines** [borgbase/vorta#1624] [FAQ] [answered]
This may be more a question about borg than vorta, but is there a way to delete files from an archive? 

Initially my virtual machines were backed up with my home directory, but the issue there is that borg is unable to deal with these virtual drives.  If the image is "modified," even by just booting the virtual machine, then the entire thing is backed up.  In the interest of having far quicker backups I have set up a separate repo & schedule to deal with just these images that runs once per week rather than more frequently.  However, all that data is now duplicated on my server (in the home dir backup before they were excluded, and now in their intended repo).  Is there a way to mount archives r/w so I can delete these? 

From what I can tell I can delete the archives that contain those files, but that will likely be a 2+ hour process (took 3.5 hours to run the initial back up) and then my guess is that I would have to run another 2+ hour backup of my home dir to back up all the other files that were deleted as well that are not contained in the later incremental archives.  

If I cannot manually delete these files on their own, my best bet is probably to just wait for the older archives to get pruned naturally.

**Comment by real-yfprojects** [comment, marked as answer]
See [borg FAQ](https://borgbackup.readthedocs.io/en/stable/faq.html#how-do-i-remove-files-from-an-existing-backup).

**Reply by benjamindaines** [reply]
Would definitely be nice to see a "recreate" option added to the archive tab.

Currently, what is the cleanest way to manually run borg recreate with my profile created in vorta?

**Reply by real-yfprojects** [reply]
> Currently, what is the cleanest way to manually run borg recreate with my profile created in vorta?

There is no clever way. You have to do it like you had been using borg from the cmd all along. You can find all the relevant info except the password in the GUI.

**Reply by real-yfprojects** [reply]
While I consider `recreate` a niche use case, this might be something worth to simplify. What do you think about adding a button for opening a terminal with the all the borg config related env vars setup?

**Reply by benjamindaines** [reply]
I like that.  Should allow advanced usage of borg without having to program every edge case into the UI.  I still think that recreate would be useful enough to folks to add as I'll often diff archives and notice more things that should be excluded, but your proposed solution would be an excellent stopgap.
